FindTeamer/FirebaseService/FirebaseAuthService.swift

```diff
@@ -33,18 +34,7 @@ final class FirebaseAuthService : AuthManager {
             completionBlock(authResult, error)
         }
     }
-    func listener() {
-        handle = Auth.auth().addStateDidChangeListener { [weak self] (_, user) in
-            guard let strongSelf = self else { return }
-            if user != nil {
-                strongSelf.isAuthorized = true
-            } else {
-                strongSelf.isAuthorized = false
-            }
-        }
-        removeListener()
-    }
-    func removeListener() {
-        Auth.auth().removeStateDidChangeListener(handle!)
+    func observeAuthenticationChanges() -> AnyPublisher<User?, Never> {
+        Publishers.AuthenticationPublisher().eraseToAnyPublisher()
     }
```

Comment:
1. Try without creating customer AuthenticationPublisher and AuthenticationSubscription.
2. You could return just boolean value if you don't need User object deeper